### PR TITLE
Add an option to have an external download/bootstrap cache

### PR DESCRIPTION
**Pull request by lu-zero**
Follow up from #116697 to address https://github.com/rust-lang/rust/pull/116697#pullrequestreview-1677176395

### CODE REVIEW

**Comment by tgross35** [comment]
Cc original author @jefferyto in case you haven't seen this (looks like they haven't been active on GH for a couple months but never hurts to check)

File: src/bootstrap/src/utils/change_tracker.rs [hunk 1]
```diff
@@ -136,4 +136,9 @@ pub const CONFIG_CHANGE_HISTORY: &[ChangeInfo] = &[
         severity: ChangeSeverity::Info,
         summary: "`x install` now skips providing tarball sources (under 'build/dist' path) to speed up the installation process.",
     },
+    ChangeInfo {
+        change_id: 121976,
+        severity: ChangeSeverity::Info,
+        summary: "A new `boostrap-cache-path` option has been introduced. Set it in your config.toml to use a different path for the download cache.",
```
Comment (onur-ozkan) [inline src/bootstrap/src/utils/change_tracker.rs:142]:
```suggestion
        summary: "A new `boostrap-cache-path` option has been introduced which can be utilized to modify the cache path for bootstrap.",
```

Comment (lu-zero) [reply src/bootstrap/src/utils/change_tracker.rs]: Thank you!

**Comment by onur-ozkan** [comment]
r? onur-ozkan

**Review by onur-ozkan — APPROVED** [review]
thank you!

**Pull request merged**